Lib/test/test_pydoc/test_pydoc.py

```diff
@@ -478,8 +478,29 @@ def test_mixed_case_module_names_are_lower_cased(self):
 
     def test_issue8225(self):
         # Test issue8225 to ensure no doc link appears for xml.etree
-        result, doc_loc = get_pydoc_text(xml.etree)
-        self.assertEqual(doc_loc, "", "MODULE DOCS incorrectly includes a link")
+        self.assertIsNone(
+            get_pydoc_link(xml.etree),
+            "MODULE DOCS incorrectly includes a link",
+            )
+
+    def test_issue40051_non_matching_doc_and_module_names(self):
+        turtledemo = import_helper.import_module('turtledemo')
+        self.assertIn('turtle.html', get_pydoc_link(turtledemo))
+
+    def test_issue40051_undocumented_modules(self):
+        self.assertIsNone(
+            get_pydoc_link(_pickle),
+            "MODULE DOCS incorrectly includes a link",
+            )
+
+    def test_issue40051_non_library_link(self):
+        du_cmd = import_helper.import_module(
+            'distutils.command',
+            deprecated=True,
+            )
+        doc_link = get_pydoc_link(du_cmd)
+        self.assertIn('distutils/apiref.html', doc_link)
```

Comment:
This example should be changed, distutils was removed

Reply:
`distutils` submodules were the only examples, so I removed the test and updated other added test names to the new issue number.